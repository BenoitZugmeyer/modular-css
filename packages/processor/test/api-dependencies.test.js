"use strict";

const namer = require("@modular-css/test-utils/namer.js");
const relative = require("@modular-css/test-utils/relative.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".dependencies()", () => {
        let processor;

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });

        it("should return the dependencies of the specified file", async () => {
            await processor.file("./packages/processor/test/specimens/start.css");

            expect(
                relative(processor.dependencies(require.resolve("./specimens/start.css")))
            )
            .toMatchSnapshot();
        });

        it("should return the overall order of dependencies if no file is specified", async () => {
            await processor.file("./packages/processor/test/specimens/start.css");

            expect(relative(processor.dependencies())).toMatchSnapshot();
        });
    });
});
