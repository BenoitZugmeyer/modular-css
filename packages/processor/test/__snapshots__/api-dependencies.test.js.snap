// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .dependencies() should return the dependencies of the specified file 1`] = `
Array [
  "packages/processor/test/specimens/folder/folder.css",
  "packages/processor/test/specimens/local.css",
]
`;

exports[`/processor.js .dependencies() should return the overall order of dependencies if no file is specified 1`] = `
Array [
  "packages/processor/test/specimens/folder/folder.css",
  "packages/processor/test/specimens/local.css",
  "packages/processor/test/specimens/start.css",
]
`;
